Source/WebKit/NetworkProcess/storage/IDBStorageRegistry.cpp

```diff
@@ -46,14 +46,18 @@ WebCore::IDBServer::IDBConnectionToClient& IDBStorageRegistry::ensureConnectionT
 
 void IDBStorageRegistry::removeConnectionToClient(IPC::Connection::UniqueID connection)
 {
-    auto allConnectionsToClient = std::exchange(m_connectionsToClient, { });
-    for (auto& [identifier, connectionToClient] : allConnectionsToClient) {
-        if (connectionToClient->ipcConnection() != connection) {
-            m_connectionsToClient.add(identifier, WTFMove(connectionToClient));
-            continue;
+    m_connectionsToClient.removeIf([&](auto& entry) {
```

Comment:
The commit log says that this PR also simplifies code in IDBStorageRegistry::removeConnectionToClient(). However, the new code is less concise somehow. Also, it doesn't seem strictly related to the fix. Do we have to do this in this patch?

Reply:
The original code creates a temp map from m_connectionsToClient for iteration and adds entries that are not removed back m_connectionsToClient after iteration; the new code just operates on m_connectionsToClient, avoiding move and assignment of values. It's more lengthy as it adds a null check.
This patch aims to fix the crash in the radar (`IDBStorageRegistry::removeConnectionToClient`), so this change is related.